sdks/go/pkg/beam/testing/ptest/ptest.go

```diff
@@ -73,7 +73,8 @@ func Run(p *beam.Pipeline) error {
 	if *Runner == "" {
 		*Runner = defaultRunner
 	}
-	return beam.Run(context.Background(), *Runner, p)
+	_, err := beam.Run(context.Background(), *Runner, p)
+	return err
```

Comment:
We should consider changing ptest or at least having a new ptest.RunWithMetrics endpoint that also returns the pipeline results, as metrics are very valuable and convenient to check in testing situations. That can be handled as a TODO (BEAM--####) however, rather than as a part of this PR, since ptest is pervasive.

Reply:
Filled https://issues.apache.org/jira/browse/BEAM-11218